(merge) Issue 4 to dev

dezmound opened the pull request:
Вливаем GraphQl в dev, вместе с конфигурацией.

JorJeG committed:
(new) Поставит отправную точку в мир бэкенда

dezmound committed:
(del) Удалит файл db из корня

Удалили файл db.sqlite3 из корня репозитория. В последствии стоит добавить его в .gitignore

dezmound committed:
(fix) Исправит связи в Sequileze

Исправил связи между сущностями Todo User Task. Теперь graphql стартует.

dezmound committed:
(new, issue-4) Добавит новые мутации в GraphQl

Добавил мутации на создание/изменение/удалене Todo/Task.

dezmound committed:
(fix, issue-4) Исправит мутации GraphQl

Испраит тип возвращаемых значений в мутациях createTodo, createTask

dezmound committed:
(fix) Исправит мутации в схеме GraphQl

Внес изменения в имена параметров в мутациях. Добавл постфикс Id к параметрам, которые являются
id-шниками.

dezmound committed:
(fix) Исправит имя свзяи пользователей и друзей

Исправил имя связи пользовать - друзья.

dezmound committed:
(new) Добавит реализацию GraphQl мутаций

Добавил базовую реализацию GraphQl мутаций.

dezmound committed:
(fix) Исправит ошибку при получении списка todo у пользоватля

Добавил связь для получения списка todo и списка task

JorJeG committed:
(fix) Исправит возвращаемые значения при мутациях

JorJeG committed:
(merge) Объеденит с правильной схемой

dezmound committed:
(config) Добавит конфигурацию сервера

Добавил директроию с конфигурационными файлами. Создал конфигурацию для подключения к БД.

JorJeG committed:
(config) Добавит правило для eslint

dezmound committed:
(fix) Исправит связи между Todos/Users/Tasks

Теперь при удалении пользователя, удалятся Todo, принадлежащие ему.
При удалении Todo удалятся Tasks, принадлежащие ему.

belodpav commented on server/app.js:
```diff
@@ -0,0 +1,19 @@
+
+const app = express();
+
+app.set('port', process.env.PORT || 3001);
```
Порт по умолчанию, наверное, стоит вынести в конфиг

belodpav replied:
тогда отпадает писать конструкции типа app.get('port') Если можно например просто брать этот порт из конфигурационного объекта.например config.PORT  или просто PORT(APP_PORT, and so on...) Может, есть какое-то обоснование по использованию app.get('port) вместо центролизованного конфига из которого поддягиваем необходимые константы приложения?

dezmound replied:
Да, все верно

belodpav commented on server/app.js:
```diff
@@ -0,0 +1,19 @@
+
+app.use('/graphql', graphqlRoutes);
+if (process.env.NODE_ENV === 'production') {
+  app.use(express.static('build'));
```
1. Вопрос: Мы подключаем статику только в 'production' режиме? А при разработке клиент-серверного приложения у нас все будет работать? 
2. Можно вынести process.env.NODE_ENV в конфиг со значением по умолчанию, тогда будет красивее и централизованнее, и уже можно будет матчится не только на 'production' но и на 'development'. Тоже самое что и с PORT. Что думаете?

dezmound replied:
Я за то, чтобы вынести `NODE_ENV` в конфиг, только сделать что то типа:
```javascript
{
    NODE_ENV_PROD: process.env.NODE_ENV === 'production',
    NODE_ENV_DEV: process.env.NODE_ENV === 'development',
}
```

dezmound replied:
Насчет окружения для статики, как минимум надо отдавать статику из `dist`, так как у нас клиент билдится туда:
```json
{
    "build": "react-scripts build && mv build ../server/dist",
}
```
И я думаю это условие все таки лишнее, на `dev` тоже статика нужна. У нас же будет `staging` в `heroku` и нет смысла на него смотреть, когда нет статики.

JorJeG replied:
Я когда делал приложение на heroku, всегда всё запускал в production.

Насчёт dev режима, статика раздаётся сервером, который запускается npm start

belodpav replied:
Давайте пока без условия сделаем. Потмо посмотрим как все будет работать. Да, кажется что оно здесь лишнее, но это не совсем явно.

belodpav commented on server/graphql/routes.js:
```diff
@@ -0,0 +1,22 @@
+
+router.use(cors(), graphqlHTTP({
+  schema,
+  graphiql: true
```
Вот здесь наврное надо поставить условие на 'development' или 'production' режим. Чтобы у нас в production режиме graphiql был false и мы не смогли бы юзать графический интерфейс страницы /grphiql/?.... это ведь только для dev режима нужно

dezmound replied:
Верное замечание, надо поправить

belodpav commented on server/graphql/typeDefs.js:
```diff
@@ -0,0 +1,82 @@
+
+input UserInput {
+    login: String!
+    avatarUrl: String!
```
В случае если у пользователь не добавил авки, Яндекс паспорт API, добавляет ссылку на дэфолтную фотку? Это к вопросу про обязательность "String!"

dezmound replied:
Да, добавит заглушку [Паспорт API](https://tech.yandex.ru/passport/doc/dg/reference/response-docpage/#norights_5)

JorJeG replied:
либо можно свою заглушку делать на основании поля в ответе "is_avatar_empty": false, но это усложнение

dezmound committed:
(config) Добавит конфигурацию сервера

Добавил конфигурационный файл сервера в конфиг приложения.

dezmound committed:
(fix) Изменит порт приложения из окружения на порт в конфиге

Изменил значение для порта приложения на значение в конфигурационном файле server.js

belodpav commented:
Оказывается, надо былы подтвердить review, я их написал утром думал, что отправил, а оказалось нет.

JorJeG commented:
Может сделаем так, что ревьюер и смерживает, чтобы процесс проходил быстрее, только ставить минимум два человека и последний делает (merge)?

dezmound committed:
(fix, config) Изменит конфигурацию приложения - окружение DEV, PROD.

Добавит переменные NODE_ENV_PROD, NODE_ENV_DEV в конфиг в зависимости от пременной окружения NODE_ENV.
Измени условия в которых была проверка значения в NODE_ENV.

dezmound committed:
(fix) Изменит директорию статики в express

Изменил директорию build на dist.

belodpav commented on server/app.js:
```diff
@@ -10,9 +10,7 @@ app.set('port', server.port);
-if (process.env.NODE_ENV === 'production') {
-  app.use(express.static('build'));
-}
+app.use(express.static('dist'));
```
В этом случае не забываем, изменить скрипт 
```
"build": "react-scripts build && mv build ../server/dist"
``` 
в файле client/package.json

belodpav commented on server/config/index.js:
```diff
@@ -2,5 +2,8 @@ const dbConfig = require('./db');
+    server: serverConfig,
+    NODE_ENV_PROD: process.env.NODE_ENV === 'production',
+    NODE_ENV_DEV: process.env.NODE_ENV === 'development' ||
+        process.env.NODE_ENV !== 'production'
```
Вот здесь не совсем понятно что происходит.
не достаточно просто написать process.env.NODE_ENV === 'development' || true?

dezmound replied:
Если у тебя NODE_ENV будет `'production'`, то это выражение вернет `true`:
```
process.env.NODE_ENV === 'development' || true
```

belodpav replied:
Тогда все четко.

JorJeG commented on server/graphql/resolvers/mutation.js:
```diff
@@ -0,0 +1,72 @@
+/* eslint-disable no-unused-vars */
```
желательно удалить

JorJeG commented on package.json:
```diff
@@ -11,7 +11,9 @@
-    "lint": "npm run stylelint && npm run lint:client && npm run lint:server"
+    "lint": "npm run stylelint && npm run lint:client && npm run lint:server",
+    "dev": "nodemon server/app.js",
+    "reset-db": "rm -f ./db.sqlite3 && touch db.sqlite3"
```
как выяснили можно удалить вторую часть с touch

dezmound committed:
(config) Исправит reset-db в pakacge.json

Удалил лишний touch в reset-db

dezmound committed:
(fix) Удалит строчку для отключения eslint в mutation.js

Удалил eslint-disable в mutation.js

belodpav commented on server/graphql/typeDefs.js:
```diff
@@ -0,0 +1,82 @@
+input TaskInput {
+    title: String!
+    isComplete: Boolean!
+
```
Пустая строчечка :)

dezmound committed:
(fix) Убрал пустую строчку в typeDefs.js

dezmound committed:
(ref) Удалит неиспользуемые параметры в query и mutation

Удлил неисопльзуемые параметры в query.js и mutation.js

JorJeG reviewed (approved):
Мержу и конфликт там не страшный

JorJeG committed:
Merge branch 'dev' into issue-4

belodpav commented on package.json:
```diff
@@ -14,6 +14,7 @@
     "lint": "npm run stylelint && npm run lint:client && npm run lint:server",
     "dev": "nodemon server/app.js",
     "reset-db": "rm -f ./db.sqlite3"
```
Запятая нужна

dezmound committed:
(fix, config) Исправит потерянную запятую

belodpav commented:
Вроде збс :smiley:

JorJeG reviewed (approved):
Всё чётко) Большой issue получился, мы справились 😉
